src/ReactART.js

```diff
@@ -261,7 +258,11 @@ var EventTypes = {
   onClick: 'click'
 };
 
-var NodeMixin = assign({}, ReactComponentMixin, {
+var NodeMixin = {
+
+  construct: function(element) {
+    this._currentElement = element;
```

Comment:
I added those expandos in the general instantiation function.

https://github.com/facebook/react/pull/2923/files#diff-c0176c2f2c4dcd1401cf97e4e767dac8R115

That way that hack is localized to one place and the extensions don't need to think about that.